sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryPipelineResponse.java

```diff
@@ -33,28 +34,39 @@ public String getBody() {
         return body;
     }
 
-    public Set<String> getErrors() {
-        return parseErrors(body); // parseErrors on demand
+    public Set<ResponseError> getErrors() {
+        try {
+            return parseErrors(body); // parseErrors on demand
+        } catch (IllegalStateException e) {
+            return emptySet();
+        }
+    }
+
+    public Set<String> getErrorMessages() {
+        Set<ResponseError> responseErrors = getErrors();
+        if (responseErrors.isEmpty()) {
+            return singleton("Could not parse response");
+        }
```

Comment:
(in case there really are no errors)
```suggestion
        Set<ResponseError> responseErrors;
        try {
            return parseErrors(body);
        } catch (IllegalStateException e) {
            return singleton("Could not parse response");
        }
```